### PR TITLE
fix deploy-scheduler-estimator.sh cannot generate correct kubeconfig file problem

### PR DESCRIPTION
**Pull request by cmicat**
fix deploy-scheduler-estimator.sh cannot generate correct kubeconfig file problem

Signed-off-by: raymondmiaochaoyue <raymondmiaochaoyue@didichuxing.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

When used as [Cluster Accurate Scheduler Estimator For Rescheduling](https://karmada.io/docs/userguide/scheduling/scheduler-estimator) guided, the original `hack/deploy-scheduler-estimator.sh` have two problems:

(1) The <MEMBER_CLUSTER_KUBECONFIG> file is not flattened, and the `kubectl create secret generic` command will copy the <MEMBER_CLUSTER_KUBECONFIG> file's content directly without changing anything. So if the <MEMBER_CLUSTER_KUBECONFIG> contains none embedded cert reference like the following, karmada-scheduler-estimator will not be able to read the certificate file. Because the certificate file is not mounted by the container.

```
client-certificate: /home/raymondmiaochaoyue/.minikube/profiles/ray-karmada-member2/client.crt
```

(2) <MEMBER_CLUSTER_KUBECONFIG> is used as secret's content, it will not be changed by `--context=` parameter in `kubectl create secret generic` command. We need change it first, then save to temporary file.

**Which issue(s) this PR fixes**:

None

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/deploy-scheduler-estimator.sh [hunk 1]
```diff
@@ -49,11 +49,23 @@ then
 fi
 MEMBER_CLUSTER_NAME=$4
 
+TEMP_PATH=$(mktemp -d)
+MEMBER_CLUSTER_KUBECONFIG_NAME=$(basename ${MEMBER_CLUSTER_KUBECONFIG})
```
Comment (ikaven1024) [inline hack/deploy-scheduler-estimator.sh:53]:
```suggestion
MEMBER_CLUSTER_KUBECONFIG_NAME=$(basename "${MEMBER_CLUSTER_KUBECONFIG}")
```

**Review by ikaven1024 — CHANGES_REQUESTED** [review]

File: hack/deploy-scheduler-estimator.sh (hunk 1 above)
Comment (cmicat) [reply hack/deploy-scheduler-estimator.sh]: > fixed

**Comment by RainbowMango** [comment]
/assign
Thanks in advance. I'll look at it tomorrow.

**Review by RainbowMango — APPROVED** [review]
Fantastic!!!
Thanks for doing this.

I don't know these flags before.
```
    --flatten=false:
	Flatten the resulting kubeconfig file into self-contained output (useful for creating portable kubeconfig
	files)
    --minify=false:
	Remove all information not used by current-context from the output
```

/lgtm
/approve

**Pull request merged**